### PR TITLE
separate apps v2 specific api from legacy api

### PR DESCRIPTION
**Pull request by anzioka**
PR:
separate apps v2 specific api from legacy api

### CODE REVIEW

**Review by makosblade — DISMISSED** [review]

File: src/main/java/com/openlattice/chronicle/api/ChronicleApi.java [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+     * @param deviceId       The device id logging the data.
+     * @param data           The data / entities to write
+     * @return The total number of items persisted by the server.
+     */
+    @POST( BASE + ORGANIZATION_ID_PATH + STUDY_ID_PATH + PARTICIPANT_ID_PATH + DATASOURCE_ID_PATH + UPLOAD_PATH )
+    Integer upload(
+            @Path( ORGANIZATION_ID ) UUID organizationId,
+            @Path( STUDY_ID ) UUID studyId,
+            @Path( PARTICIPANT_ID ) String participantId,
+            @Path( DATASOURCE_ID ) String deviceId,
```
Comment (UnsungHero97) [inline src/main/java/com/openlattice/chronicle/api/ChronicleApi.java:278]: Is it a data source id or a device id. Let's be consistent (i.e. data source).

**Review by UnsungHero97 — CHANGES_REQUESTED** [review]

File: src/main/java/com/openlattice/chronicle/api/ChronicleApi.java [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+     * @return The internal chronicle id for a device. It can be used to track a single device across resets, app uninstalls,
+     * etc.
+     */
+
+    @POST( BASE + ORGANIZATION_ID_PATH + STUDY_ID_PATH + PARTICIPANT_ID_PATH + DATASOURCE_ID_PATH + ENROLL_PATH )
+    UUID enrollSource(
+            @Path( ORGANIZATION_ID ) UUID organizationId,
+            @Path( STUDY_ID ) UUID studyId,
+            @Path( PARTICIPANT_ID ) String participantId,
+            @Path( DATASOURCE_ID ) String datasourceId,
```
[the remark this answers is not in the document]
Comment (anzioka) [reply src/main/java/com/openlattice/chronicle/api/ChronicleApi.java:74]: The Datasource object is an `Optional`

File: src/main/java/com/openlattice/chronicle/api/ChronicleApi.java [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+
+    /**
+     * Delete a participant and their data.  Returns the number of entities removed.
+     *
+     * @param organizationId - Id of the organization to which study belongs
+     * @param studyId        - studyId
+     * @param participantId  - participant id
+     */
+    @DELETE( BASE + AUTHENTICATED + ORGANIZATION_ID_PATH + STUDY_ID_PATH + PARTICIPANT_ID_PATH )
+    Void deleteParticipantAndAllNeighbors(
```
[the remark this answers is not in the document]
Comment (anzioka) [reply src/main/java/com/openlattice/chronicle/api/ChronicleApi.java:85]: Why would that be problematic?

File: src/main/java/com/openlattice/chronicle/api/ChronicleApi.java [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+     * Returns a file download containing all participant data (including neighbor data).
+     *
+     * @param organizationId         - Id of the organization to which study belongs
+     * @param studyId                - the study id
+     * @param participantEntityKeyId - the participant entity key id
+     * @param fileType               - the type of file (csv, json) to return as the download
+     * @return All participant data
+     */
+    @GET( BASE + AUTHENTICATED + PARTICIPANT_PATH + DATA_PATH + ORGANIZATION_ID_PATH + STUDY_ID_PATH
+            + ENTITY_KEY_ID_PATH )
```
[the remark this answers is not in the document]
Comment (anzioka) [reply src/main/java/com/openlattice/chronicle/api/ChronicleApi.java:115]: The current pattern is consistent with the legacy api

File: src/main/java/com/openlattice/chronicle/api/ChronicleApi.java [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+    /**
+     * Returns a file download containing preprocessed data.
+     *
+     * @param organizationId         - Id of the organization to which study belongs
+     * @param studyId                - the study id
+     * @param participantEntityKeyId - the participant entity key id
+     * @param fileType               - the type of file (csv, json) to return as the download
+     * @return All participant data
+     */
+    @GET( BASE + AUTHENTICATED + PARTICIPANT_PATH + DATA_PATH + ORGANIZATION_ID_PATH + STUDY_ID_PATH
```
[the remark this answers is not in the document]
Comment (anzioka) [reply src/main/java/com/openlattice/chronicle/api/ChronicleApi.java:132]: https://github.com/openlattice/chronicle-api/pull/38#discussion_r514264380

File: src/main/java/com/openlattice/chronicle/api/ChronicleApi.java [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+    /**
+     * Returns a file download containing user app usage data.
+     *
+     * @param organizationId         - Id of the organization to which study belongs
+     * @param studyId                - the study id
+     * @param participantEntityKeyId - the participant entity key id
+     * @param fileType               - the type of file (csv, json) to return as the download
+     * @return All participant data
+     */
+    @GET( BASE + AUTHENTICATED + PARTICIPANT_PATH + DATA_PATH + ORGANIZATION_ID_PATH + STUDY_ID_PATH
```
[the remark this answers is not in the document]
Comment (anzioka) [reply src/main/java/com/openlattice/chronicle/api/ChronicleApi.java:150]: https://github.com/openlattice/chronicle-api/pull/38#discussion_r514264380

File: src/main/java/com/openlattice/chronicle/api/ChronicleApi.java [hunk 7]
```diff
@@ -0,0 +1,292 @@
...
+            @Body List<SetMultimap<UUID, Object>> data );
+
+    /**
+     * Looks up property type ids for the requested property type FQNS (passed as Strings)
+     *
+     * @param propertyTypeFqns A set of property type FQNs (as Strings)
+     * @return A map from property type FQNs to their corresponding ids
+     */
+    @POST( BASE + EDM_PATH )
+    Map<String, UUID> getPropertyTypeIds( @Body Set<String> propertyTypeFqns );
```
[the remark this answers is not in the document]
Comment (anzioka) [reply src/main/java/com/openlattice/chronicle/api/ChronicleApi.java:288]: We could, but that sounds like unnecessary overhead work on the android client

File: src/main/java/com/openlattice/chronicle/api/ChronicleApi.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (UnsungHero97) [reply src/main/java/com/openlattice/chronicle/api/ChronicleApi.java]: Something about Spring truncating `@PathVariable` strings with dots...

https://www.baeldung.com/spring-mvc-pathvariable-dot

> Spring considers that anything behind the last dot is a file extension

Look at `PrincipalApi` for how we support email searches.

File: src/main/java/com/openlattice/chronicle/api/ChronicleApi.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (UnsungHero97) [reply src/main/java/com/openlattice/chronicle/api/ChronicleApi.java]: Why are you trying to be consistent with the legacy api?

**Review by UnsungHero97 — APPROVED** [review]

**Review by giuliacampana — APPROVED** [review]

**Pull request merged**